pkg/controller/launcher/job.go

```diff
@@ -239,6 +243,41 @@ func getBatchJob(clusterName string, imageURI string, curator clustercuratorv1.C
 				},
 			},
 		}
+	case "installPosthook", "upgradePosthook":
+		if (desiredCuration == "installPosthook" && curator.Spec.Install.Posthook != nil) || (desiredCuration == "upgradePosthook" && curator.Spec.Upgrade.Posthook != nil) {
+			isPosthook = true
+		}
+		newJob = &batchv1.Job{
+			ObjectMeta: v1.ObjectMeta{
+				GenerateName: "curator-job-",
+				Namespace:    clusterName,
+				Labels: map[string]string{
+					"open-cluster-management": "curator-job",
+				},
+				Annotations: map[string]string{
+					PostAJob:     "Retry an install posthook job",
+					DoneDoneDone: "Cluster Curator job has completed",
+				},
+			},
+			Spec: batchv1.JobSpec{
+				BackoffLimit:            new(int32),
+				TTLSecondsAfterFinished: &ttlf,
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						ServiceAccountName: "cluster-installer",
+						RestartPolicy:      corev1.RestartPolicyNever,
+						InitContainers:     []corev1.Container{},
+						Containers: []corev1.Container{
+							corev1.Container{
+								Name:    DoneDoneDone,
+								Image:   imageURI,
+								Command: append([]string{CurCmd, DoneDoneDone}),
```

Comment:
I am still not too familiar with this code but DoneDoneDone launches post hooks?

Reply:
Nope :) the DoneDoneDone container only runs when all other jobs are successful. It does cleanup like remove CuratingJob, Status from the curator resource.

The isPosthook flag at https://github.com/stolostron/cluster-curator-controller/pull/159/files#diff-bf83b388c4c87db8c671f7e5728bec15a9e587d04bf147cb7d061c5b585bb0c5R248 tells the curator whether to run the posthook.

Then code at https://github.com/stolostron/cluster-curator-controller/blob/main/pkg/controller/launcher/job.go#L267 will add a posthook init container to the job.